### PR TITLE
New ee place skill

### PR DESCRIPTION
**Pull request by Achuthankrishna**
This branch consists of code changes for incorporating new semantic place ee skill into main, which was tested in lab and also in apartment with nav-pick-nav-place . The results can be found using this [link ](https://docs.google.com/spreadsheets/d/18cxLUSRR4FSJ3yffSowWsWS4hxU-mPm2jQheeB8TyK8/edit?usp=sharing).

### CODE REVIEW

File: spot_rl_experiments/spot_rl/envs/base_env.py [hunk 1]
```diff
@@ -1137,6 +1137,14 @@ def turn_wrist(self):
         self.spot.set_arm_joint_positions(positions=arm_positions, travel_time=0.3)
         time.sleep(0.6)
 
+    # New function to turn wrist to clip it to 160 deg (as per BD Documentation)
```
Comment (joannetruong) [inline spot_rl_experiments/spot_rl/envs/base_env.py:1140]: We can move this to place_env.py within the SemanticPlaceEE class

File: spot_rl_experiments/spot_rl/envs/base_env.py [hunk 2]
```diff
@@ -1137,6 +1137,14 @@ def turn_wrist(self):
...
+    def turn_wrist_place(self):
+        arm_positions = np.array(self.current_arm_pose)
+        arm_positions[-1] = arm_positions[-1] + np.deg2rad(90)
```
Comment (joannetruong) [inline spot_rl_experiments/spot_rl/envs/base_env.py:1143]: arm_positions[-1] += np.deg2rad(90)

File: spot_rl_experiments/spot_rl/envs/place_env.py [hunk 3]
```diff
@@ -100,14 +100,22 @@ def __init__(self, config, spot: Spot):
         # Place steps
         self._time_step = 0
 
+        # Check whether its ee env
+        self.ee_env = False
```
Comment (joannetruong) [inline spot_rl_experiments/spot_rl/envs/place_env.py:104]: remove

File: spot_rl_experiments/spot_rl/envs/place_env.py [hunk 4]
```diff
@@ -100,14 +100,22 @@ def __init__(self, config, spot: Spot):
...
         """Decide the place location"""
 
         # User does not set the gripper orientation
         if ee_orientation_at_grasping is None:
-            self.initial_arm_joint_angles = np.deg2rad(
-                self.config.INITIAL_ARM_JOINT_ANGLES_SEMANTIC_PLACE
-            )
+            if self.is_ee_env:
+                self.initial_arm_joint_angles = np.deg2rad(
+                    self.config.INITIAL_ARM_JOINT_ANGLES_SEMANTIC_PLACE_EE
```
Comment (joannetruong) [inline spot_rl_experiments/spot_rl/envs/place_env.py:113]: let's make a `decide_init_arm_joint` function in `SpotSemanticPlaceEEEnv` to override this

File: spot_rl_experiments/spot_rl/envs/place_env.py [hunk 5]
```diff
@@ -164,6 +172,10 @@ def step(self, action_dict: Dict[str, Any], *args, **kwargs):
         # Place command is issued if the place action is smaller than zero
         place = action_dict.get("grip_action", None) <= 0.0
 
+        if place:
+            self.turn_wrist_place()
```
Comment (joannetruong) [inline spot_rl_experiments/spot_rl/envs/place_env.py:176]: let's make a new `step` function in `SpotSemanticPlaceEEEnv` to override this

**Review by joannetruong — CHANGES_REQUESTED** [review]
Let's create new functions in `SpotSemanticPlaceEEEnv` to include the changes instead of modifying `SpotSemanticPlaceEnv` directly

**Comment by Achuthankrishna** [comment]
This PR is merged into main with #244 . Closing this PR.

**Pull request closed without merging**